### PR TITLE
docs: Cleanup (and fix) docker installation instructions

### LINKED ISSUE #15730: Show non-dev image installation on  getting-started/install page
### description

currently in the docker section on the install page it shows instructions for installing the dev image not the stable image

eg here https://www.envoyproxy.io/docs/envoy/v1.17.1/start/install#install-envoy-using-docker

whereas the getenvoy images point to stable 

this is kinda inconsistent and confusing and its not clear in historical docs which image it is pointing to

i would propose to add another section for installing dev images and make sure the install section shows instructions for installing latest stable

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: docs: Cleanup (and fix) docker installation instructions
Additional Description:

Currently the install page shows docker images that dont exist. This PR will fix that

It will also fix #15730 - ie make clear in all docs how to run latest stable and current dev

In addition it shifts the windows docker install instructions into the docker section and out of the host install sections (and thus bringing parity)

Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
docs rendered here - https://storage.googleapis.com/envoy-pr/16380/docs/index.html

**Comment by phlax** [comment]
cc @mattklein123 @davinci26

File: docs/root/start/install.rst
Comment (davinci26) [inline docs/root/start/install.rst:175]: What is the value of this warning? Windows is going GA, next week and the code is already complete. I would remove it

**Review by davinci26 — CHANGES_REQUESTED** [review]
Thanks for the cleanup. LGTM minus one small comment

File: docs/root/start/install.rst
Comment (phlax) [reply docs/root/start/install.rst:175]: 2 things:

- afaict these images are experimental - and should be treated as such
- in an open source project documenting and/or distributing binary blobs should come with a warning

this is "bringing parity" - i would add similar warnings to any other code/images etc

Comment (phlax) [reply docs/root/start/install.rst:175]: >  Why do you think they are experimental? 

at least afaict they have reduced functionality and are less performant - if these images are genuinely production-ready we can remove this bit - but im already seeing "it doesnt work" tickets, so my concern is both to be explicit about the fact that they are not widely tested and to add a "caveat usor"

> If this is the case shouldn't all images have this warning then?

the other images *are* open source - they dont need it - there is no code in them for which the source is not available - and the recipes are published

Comment (phlax) [reply docs/root/start/install.rst:175]: ... and the user can compile the entire  stack for themselves should they wish

Comment (phlax) [reply docs/root/start/install.rst:175]: s/distributing binary blobs/distributing closed-source binary blobs/

Comment (davinci26) [reply docs/root/start/install.rst:175]: That is a good opportunity to clarify the ambiguities.

> at least afaict they have reduced functionality and are less performant

re feature parity: There are some features that are missing which we explicitly document but neither of the missing features is deemed high priority from customers.

re performance: This is not actually true. We have comparable median TCP termination performance on Windows and somewhat reduced p95 performance. There were some issues in the alpha release regarding performance which we addressed.

re tickets: I have not seen a production bug yet but I have seen questions regarding usability. This is an opportunity for us to improve the documentation. We already have teams within msft that are adopting Envoy on Windows in production.

Comment (davinci26) [reply docs/root/start/install.rst:175]: Regarding the disclaimer for the open source code I will defer to @wrowe to chime in here

Comment (davinci26) [reply docs/root/start/install.rst:175]: I have seen this and I have this on my radar. This is a question for now.

Comment (phlax) [reply docs/root/start/install.rst:175]: also worth pointing out the *compressed* size windows image is 2.3GB (ie the compressed size of the not open-source binary blobs), whereas the linux equivalent is 50MB

even if this is "production-ready" its not afaict suitable for production use - esp not in the situation that Envoy is most typically deployed in - ie as a sidecar

Comment (antoniovicente) [reply docs/root/start/install.rst:175]: @envoyproxy/windows-dev 

Would it be possible to get more context about closed source binaries included in the windows image and where they come from?

@wrowe and Windows Envoy maintainers: what would be your preferred wording for this section?  Worth adding some notes about size and plans to improve on that?

Comment (sunjayBhatia) [reply docs/root/start/install.rst:175]: The Windows Server Core base image usually comes pre-cached on a lot of cloud provided images etc., so its size is an issue but the impact can be minimized. Definitely already being used in production in other products (cloudfoundry, etc.). It should consist of 2 layers, the base that is tied to the host version, and another layer with Windows updates that gets updated periodically. B/c of the host/container image compatibility requirements now, the large base layer won't change (much/ ever) I believe in the lifecycle of a Windows version.

Comment (sunjayBhatia) [reply docs/root/start/install.rst:175]: Also the base container image for Windows is not being distributed by Envoy/DockerHub per-se, the Windows proprietary/licensed layers do not get hosted on DockerHub, they are foreign layers that still come from Microsoft when you pull the image (the image manifest should show this) that have their own license deal. I think the Envoy project is only on the hook for the layers that we build on top of the Windows base.

Comment (davinci26) [reply docs/root/start/install.rst:175]: > Worth adding some notes about size and plans to improve on that?

The plans are related to reducing the size of the base image. I do not think that Envoy docs is the best channels to communicate such changes though.

File: docs/root/start/install.rst
Comment (wrowe) [inline docs/root/start/install.rst:178]: The first statement is wrong, the port is code complete, and this has been discussed by the maintainers, at the community meeting. Please fix.

The second statement is confusing at best; neither the Windows image nor the OS-free glibc image are exactly satisfied by our Apache License 2.0. A valuable statement here with respect to each docker installation would be a footnote (not "warning") that Windows includes the respective base operating systems (Windows LTSC2019 - long term server core of Server 2019, or alpine-3.12_glibc-2.31, or google vrp context) under their respective licensing. This [link](https://hub.docker.com/_/microsoft-windows-base-os-images) offers the necessary licensing context for the MS Windows docker base image.

Please don't use envoy docs for personal evangelism. Both Windows and Linux docker users locally and users of public clouds are generally aware of what their docker or host licensing prerequisites entail.

**Review by wrowe — CHANGES_REQUESTED** [review]
This is looking good, especially the build cleanups and docs simplification. Just a few semantic problems to fix that were already discussed at length, my requested changes below...

File: docs/root/start/install.rst
Comment (wrowe) [reply docs/root/start/install.rst:137]: Actually, I'd rephrase this, as we have thin Docker images applicable to any linux. It's not necessary to place that footnote in every linux instruction, perhaps something topline they would read before scrolling to their target linux flavor. Once they've read their target instructions, it seems likely they could miss the later section for installing with docker.

**Comment by phlax** [comment]
reading the comments above im hearing a lot of appeals for special casing (and some borderline insults)

the facts as i understand:

- this image is not open source - unlike everything else distributed within CNCF and envoy namespace
- the image is huge - everywhere else that a huge (ie > 1GB) image is documented it comes with a warning 
- the image is not well-tested - it had ~170 downloads last time i checked, comapred to >100 million for the envoy linux image

i understand the desire for parity - but that means playing by the same rules as everyone else

my own opinion is that this image should not be distributed within the CNCF namespace as it is misleading to end users (not to mention the increased maintenance burden that cant be checked/supported if you dont use closed-source code)

regardless, i have done my best to accommodate - which (i hope) @davinci26 will attest

currently the windows **docker** install section is placed amongst the host installs - it needs to move to the docker section

this pr also provides a bunch of ~unrelated cleanups regarding the publishing of docker images

whatever my own opinion - im trying to bring actual parity to the documentation, and that means giving users fair warning when they are being encouraged to download binaries which are closed, huge or relatively untested

**Review by wrowe — CHANGES_REQUESTED** [review]

File: docs/root/start/install.rst
Comment (mattklein123) [reply docs/root/start/install.rst:175]: Let's discuss in our meeting tomorrow morning, but in general my view is:
1) Remove the experimental warning. We deemed it GA and users understand when it was GA, can open bugs, etc.
2) Remove the not open source warning (Envoy itself is Apache 2, and any base image is owned by Microsoft, and anyone using Windows understands what they are dealing with). If there are legal concerns here we can run them by the CNCF.
3) Optionally add a warning about size and link to the tracking issue, though I tend to agree with @wrowe that anyone that downloads the image knows it's big. There may be value though in telling people the tracking issue.

File: docs/build.sh
Comment (mattklein123) [inline docs/build.sh:18]: Sorry can you explain this part more? I'm confused. Shouldn't the docs just be snapped to the versions that are going to be published? I don't think this should be the last version?

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Thanks for the cleanups here. A few questions/comments.

/wait

File: docs/root/start/install.rst
Comment (wrowe) [reply docs/root/start/install.rst:178]: Reminder that we agreed to drop the statement above in line 177.

**Review by wrowe — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
